example_test.go

```diff
@@ -0,0 +1,46 @@
+package quamina_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/timbray/quamina"
+)
+
+const userRegisteredEvent = `{
+  "id": "1c0e1ce4-3d88-4786-a09d-7133c170d02a",
+  "type": "UserRegistered",
+  "user": {
+    "name": "Doe, John",
+    "premiumAccount": true
+  }
+}
+`
+
+const premiumUserPattern = `{
+  "type":["UserRegistered"],
+  "user": {"premiumAccount": [true]}
+}`
+
+func ExampleNew() {
+	q, err := quamina.New()
+	if err != nil {
+		log.Fatalf("could not create quamina instance: %v", err)
+	}
+
+	err = q.AddPattern("premium user", premiumUserPattern)
+	if err != nil {
+		log.Fatalf("could not add pattern: %v", err)
+	}
+
+	event, err := q.MatchesForEvent([]byte(userRegisteredEvent))
+	if err != nil {
+		log.Fatalf("could not match for event: %v", err)
+	}
+
+	if event != nil {
```

Comment:
I'm pretty sure that Quamina never returns nil. It always returns []X which of course might be empty.  The kind of check you want to do is something like 
```go
if len(matches) == 0 || matches[0] != "premium user" {
  t.Error(whatever)
}
```

Reply:
Ah! It's even in the docs. I just followed go "idioms" here, but you're right: it preallocates a slice!

Reply:
Yes, that makes the calling sequence a little simpler for the caller, no need to check for nil.

Reply:
I slightly improved it for better readability/to follow along the example.